### PR TITLE
Cleaned up some emag_acts

### PR DESCRIPTION
**Pull request by ShizCalev**

### CODE REVIEW

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -459,7 +459,7 @@
 					dat += "<BR>\[ <A HREF='?src=\ref[src];operation=changeseclevel'>Change Alert Level</A> \]"
 					dat += "<BR>\[ <A HREF='?src=\ref[src];operation=emergencyaccess'>Emergency Maintenance Access</A> \]"
 					dat += "<BR>\[ <A HREF='?src=\ref[src];operation=nukerequest'>Request Nuclear Authentication Codes</A> \]"
-					if(src.emagged == 0)
+					if(!src.emagged)
```
Comment (lzimann) [inline code/game/machinery/computer/communications.dm]: remove `src`

File: code/game/mecha/mech_fabricator.dm [hunk 2]
```diff
@@ -84,19 +84,16 @@
 	return 1
 
 /obj/machinery/mecha_part_fabricator/emag_act()
-	if(emagged)
```
Comment (lzimann) [inline code/game/mecha/mech_fabricator.dm]: I honestly prefer this  the old way. Just make the `emagged = FALSE` be a `TRUE` and remove the last line.

File: code/game/machinery/buttons.dm [hunk 3]
```diff
@@ -103,9 +103,11 @@
 		return ..()
 
 /obj/machinery/button/emag_act(mob/user)
-	req_access = list()
-	req_one_access = list()
-	playsound(src.loc, "sparks", 100, 1)
+	if(!emagged)
+		req_access = null
+		req_one_access = null
+		playsound(src.loc, "sparks", 100, 1)
```
Comment (Cyberboss) [inline code/game/machinery/buttons.dm]: playsound(src,...

File: code/game/machinery/computer/apc_control.dm [hunk 4]
```diff
@@ -192,13 +192,12 @@
...
-		return
-	user.visible_message("<span class='warning'>You emag [src], disabling precise logging and allowing you to clear logs.</span>")
-	log_game("[key_name_admin(user)] emagged [src] at [get_area(src)], disabling operator tracking.")
-	playsound(src, "sparks", 50, 1)
-	emagged = TRUE
+/obj/machinery/computer/apc_control/emag_act(mob/user)
+	if(!emagged)
+		user.visible_message("<span class='warning'>You emag [src], disabling precise logging and allowing you to clear logs.</span>")
+		log_game("[key_name_admin(user)] emagged [src] at [get_area(src)], disabling operator tracking.")
+		playsound(src.loc, "sparks", 50, 1)
```
Comment (Cyberboss) [inline code/game/machinery/computer/apc_control.dm]: playsound(src,...

File: code/game/machinery/limbgrower.dm [hunk 5]
```diff
@@ -233,10 +232,9 @@
...
-	for(var/datum/design/D in files.possible_designs)
-		if((D.build_type & LIMBGROWER) && ("special" in D.category))
-			files.AddDesign2Known(D)
-	to_chat(user, "A warning flashes onto the screen, stating that safety overrides have been deactivated")
-	emag = TRUE
+	if(!emagged)
+		for(var/datum/design/D in files.possible_designs)
+			if((D.build_type & LIMBGROWER) && ("special" in D.category))
+				files.AddDesign2Known(D)
+		to_chat(user, "A warning flashes onto the screen, stating that safety overrides have been deactivated")
```
Comment (Cyberboss) [inline code/game/machinery/limbgrower.dm]: Give this a warning span

File: code/game/machinery/recycler.dm [hunk 6]
```diff
@@ -195,7 +195,7 @@
 	// For admin fun, var edit emagged to 2.
 	if(gib || emagged == 2)
 		L.gib()
-	else if(emagged == 1)
+	else if(emagged == TRUE)
```
Comment (Cyberboss) [inline code/game/machinery/recycler.dm]: this one needs to be reverted

File: code/game/mecha/mech_fabricator.dm [hunk 7]
```diff
@@ -84,19 +84,16 @@
...
+	if(!emagged)
+		say("DB error \[Code 0x00F1\]")
+		sleep(10)
+		say("Attempting auto-repair...")
+		sleep(15)
+		say("User DB corrupted \[Code 0x00FA\]. Truncating data structure...")
+		sleep(30)
+		say("User DB truncated. Please contact your Nanotrasen system operator for future assistance.")
+		req_access = null
+		emagged = TRUE
```
Comment (Cyberboss) [inline code/game/mecha/mech_fabricator.dm]: put these two lines at the top and use timers for the sleeps

File: code/modules/atmospherics/machinery/airalarm.dm [hunk 8]
```diff
@@ -711,11 +711,10 @@
...
 
 /obj/machinery/airalarm/emag_act(mob/user)
-	if(emagged)
-		return
-	emagged = TRUE
-	visible_message("<span class='warning'>Sparks fly out of the [src]!</span>", "<span class='notice'>You emag the [src], disabling its safeties.</span>")
-	playsound(src.loc, 'sound/effects/sparks4.ogg', 50, 1)
+	if(!emagged)
+		emagged = TRUE
+		visible_message("<span class='warning'>Sparks fly out of the [src]!</span>", "<span class='notice'>You emag the [src], disabling its safeties.</span>")
```
Comment (Cyberboss) [inline code/modules/atmospherics/machinery/airalarm.dm]: Remove the `the`s in front of the `src`s

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/mecha/mech_fabricator.dm (hunk 2 above)
Comment (ShizCalev) [inline code/game/mecha/mech_fabricator.dm]: Done

File: code/game/machinery/vending.dm [hunk 9]
```diff
@@ -384,7 +384,7 @@
 
 /obj/machinery/vending/emag_act(mob/user)
 	if(!emagged)
-		emagged  = 1
+		emagged = 1
```
Comment (Cyberboss) [inline code/game/machinery/vending.dm]: TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/mecha/mech_fabricator.dm (hunk 2 above)
Comment (Cyberboss) [inline code/game/mecha/mech_fabricator.dm]: he wants it like
```
if(emagged)
    return
etc...
```
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-early-return

**Pull request merged**